sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/toFromProtobufSqlFunctions.scala

```diff
@@ -88,6 +88,26 @@ case class FromProtobuf(
     messageName: Expression,
     descFilePath: Expression,
     options: Expression) extends QuaternaryExpression with RuntimeReplaceable {
+
+  def this(data: Expression, messageName: Expression, descFilePathOrOptions: Expression) = {
+    this(
+      data,
+      messageName,
+      descFilePathOrOptions.dataType match {
```

Comment:
Should we assert this is a literal? The problem with using `dataType` directly is that an unresolved expression will throw an exception if you touch it.

Reply:
Sure, let me update it